### DISCUSSION TITLE
How do I configure the package manager version to be used?

### DISCUSSION THREAD

**Discussion by Kircheneer** [renovatebot/renovate#16785] [Request Help] [answered]
Poetry has a [bug](https://github.com/python-poetry/poetry/issues/6045) where `poetry.lock` has a bunch of empty lists for the `metadata.files` key (see example of renovate exhibiting this behavior [here](https://github.com/networktocode/diffsync/pull/138/files)). According to [this](https://docs.mend.io/bundle/integrations/page/mend_remediate_package_managers_release_notes.html) the current version of Poetry that renovate uses is `1.1.13` which is one patch version away from `1.1.14` which fixes this issue.

So the question is - how can I configure renovate to user another version than this one? I wasn't able to find the setting in the documentation.

**Comment by viceice** [comment, marked as answer]
https://github.com/renovatebot/renovate/blob/f1bcafc5840f73716a668ef99b04fcbed494817d/lib/modules/manager/poetry/artifacts.ts#L50-L79

this code is used to determine poetry version to use, otherwise renovate should use latest version from pypi

https://github.com/renovatebot/renovate/blob/f1bcafc5840f73716a668ef99b04fcbed494817d/lib/util/exec/buildpack.ts#L72-L76

**Reply by viceice** [reply]
you can use `constraints` to force another version

https://docs.renovatebot.com/configuration-options/#constraints
